### PR TITLE
{AKS} `az aks update`: Fix aks network profile update error

### PR DESCRIPTION
**Pull request by MartinForReal**
**Related command**

**Description**

fix regression bug introduced in #26975

**Testing Guide**

**History Notes**

[AKS] `az aks update`: Fix aks network profile update error

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py [hunk 1]
```diff
@@ -2065,7 +2065,7 @@ def _get_outbound_type(
         outbound_type = self.raw_param.get("outbound_type")
         # try to read the property value corresponding to the parameter from the `mc` object
         read_from_mc = False
-        if self.decorator_mode == DecoratorMode.CREATE:
+        if self.decorator_mode != DecoratorMode.CREATE:
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py:2068]: After the change, under update scenario, won't the value specified by user be overwritten by the existing value from mc?

Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py:2068]: Queued test [test_aks_update_outbound_from_slb_to_natgateway](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=77089559&view=logs&j=b162b355-d59d-5864-ce0f-0a70f12dd28b&t=dc59ccd1-231f-538b-777f-33a592c7ca57&l=408), seems the update won't work

Comment (MartinForReal) [reply src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py]: Updated the commit and queued teshttps://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=77091936&view=results

Comment (MartinForReal) [reply src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py]: And it worked

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**